From 19e0e077a1232e62ab08ccafb0180abf10b0414e Mon Sep 17 00:00:00 2001
From: David Clark <dclark@sierrawireless.com>
Date: Tue, 12 Dec 2017 15:05:28 -0800
Subject: [PATCH] Continue retrying block when 408 error is received

Resolves: LE-8894
Change-Id: Ic694d57f08cc5b32925bb5ae7876ca7181f00e4f
Signed-off-by: David Clark <dclark@sierrawireless.com>
---

diff --git a/core/packet.c b/core/packet.c
index ec7ddcf..e0e77d7 100644
--- a/core/packet.c
+++ b/core/packet.c
@@ -671,6 +671,11 @@
                     LOG("All blocks transferred.");
                     prv_end_push(push_stateP);
                 }
+                else if (message->code == COAP_408_REQ_ENTITY_INCOMPLETE)
+                {
+                    LOG("Incomplete block transfer.");
+                    return;
+                }
 #endif
                 transaction_handleResponse(contextP, fromSessionH, message, NULL);
                 break;
