From d7a8fb1c5b36073099daf108ca0077ea91027965 Mon Sep 17 00:00:00 2001
From: David Clark <dclark@sierrawireless.com>
Date: Tue, 12 Dec 2017 08:54:25 -0800
Subject: [PATCH] Only re-authenticate before tx blk 1 of a message

Resolves: LE-8891
Change-Id: Ibcfbe2aae8914edd6554c3a3a4f534cca0f5ed1e
---

diff --git a/sessionManager/dtlsConnection.c b/sessionManager/dtlsConnection.c
index 1e2e694..e75e3e5 100644
--- a/sessionManager/dtlsConnection.c
+++ b/sessionManager/dtlsConnection.c
@@ -880,7 +880,8 @@
 (
     dtls_Connection_t* connPtr,         ///< [IN] DTLS connection structure
     uint8_t* bufferPtr,                 ///< [IN] Buffer to be sent
-    size_t length                       ///< [IN] Buffer length
+    size_t length,                      ///< [IN] Buffer length
+    bool firstBlock                     ///< [IN] First data block
 )
 {
     if (NULL == connPtr->dtlsSessionPtr)
@@ -897,7 +898,8 @@
     {
         time_t timeFromLastData = lwm2m_gettime() - connPtr->lastSend;
         LOG_ARG("now - connP->lastSend %d", timeFromLastData);
-        if ((0 < DTLS_NAT_TIMEOUT)
+        if (firstBlock
+         && (0 < DTLS_NAT_TIMEOUT)
          && ((DTLS_NAT_TIMEOUT < timeFromLastData)
             // If difference is negative, a time update could have been made on platform side.
             // In this case, do a rehandshake
@@ -1023,7 +1025,8 @@
     void* sessionHPtr,      ///< [IN] Session handle identifying the peer (opaque to the core)
     uint8_t* bufferPtr,     ///< [IN] Data to be sent
     size_t length,          ///< [IN] Data length
-    void* userDataPtr       ///< [IN] Parameter to lwm2m_init()
+    void* userDataPtr,      ///< [IN] Parameter to lwm2m_init()
+    bool firstBlock         ///< [IN] First block
 )
 {
     dtls_Connection_t* connPtr = (dtls_Connection_t*) sessionHPtr;
@@ -1036,7 +1039,7 @@
         return COAP_500_INTERNAL_SERVER_ERROR ;
     }
 
-    if (-1 == ConnectionSend(connPtr, bufferPtr, length))
+    if (-1 == ConnectionSend(connPtr, bufferPtr, length, firstBlock))
     {
         LOG_ARG("#> failed sending %lu bytes", length);
         return COAP_500_INTERNAL_SERVER_ERROR ;
