diff --git a/service/platformAdaptor/ti/pa_wifi_client_ti.c b/service/platformAdaptor/ti/pa_wifi_client_ti.c
index b799119..bcd77b2 100644
--- a/service/platformAdaptor/ti/pa_wifi_client_ti.c
+++ b/service/platformAdaptor/ti/pa_wifi_client_ti.c
@@ -565,7 +565,7 @@ le_result_t pa_wifiClient_GetScanResult
 
         if (0 == strncmp("\tSSID: ", path, 7))
         {
-            accessPointPtr->ssidLength = strnlen(path, LE_WIFIDEFS_MAX_SSID_BYTES) - 7 - 1;
+            accessPointPtr->ssidLength = strnlen(path, LE_WIFIDEFS_MAX_SSID_BYTES + 7) - 7 - 1;
             LE_INFO("FOUND SSID: '%s'", &path[7]);            
             memcpy(&accessPointPtr->ssidBytes, &path[7], accessPointPtr->ssidLength);
             LE_INFO("SSID: '%s'", &accessPointPtr->ssidBytes[0]);
