diff --git a/avcDaemon/avcServer.c b/avcDaemon/avcServer.c
index 3cc8a40..0040e9b 100644
--- a/avcDaemon/avcServer.c
+++ b/avcDaemon/avcServer.c
@@ -1500,8 +1500,8 @@ static void ProcessUpdateStatus
             break;
 
         default:
-            LE_DEBUG("Unhandled updateStatus %s", AvcSessionStateToStr(data->updateStatus));
-            break;
+            LE_WARN("Unhandled updateStatus %s", AvcSessionStateToStr(data->updateStatus));
+            return;
     }
 
     // Process user agreement or forward to control app if applicable.
